src/slackapp.js

```diff
@@ -65,71 +65,30 @@ function setupApp(slackapp, config, businessUnitProvider, trustpilotApi) {
   }
 
   function askForReply(bot, message) {
+    var originalTs = message.original_message.ts;
     var reviewId = message.original_message.attachments[0].callback_id;
-    var replyStepMessage = message.original_message;
-    replyStepMessage.text = 'You are replying to';
-    replyStepMessage.attachments[0].actions = null;
-    bot.replyInteractive(message, replyStepMessage);
-
-    bot.replyInThread(replyStepMessage, {
-      'text': `Please write your reply in this thread, in as many lines as you need. Hit the "Send reply" button
- when you're done.`,
-      'attachments': [{
-        'callback_id': reviewId,
-        'attachment_type': 'default',
-        'text': '',
-        'actions': [{
-          'name': 'step_2_send_reply',
-          'text': ':postal_horn: Send reply',
-          'value': 'step_2_send_reply',
-          'type': 'button'
-        }]
-      }]
-    });
-  }
+    var dialog = bot.createDialog('Reply to a review', JSON.stringify({originalTs, reviewId}), 'Send')
+      .addTextarea('Your reply', 'reply');
 
-  function collectUserMessages(bot, user, channel, threadTs) {
-    bluebird.promisifyAll(bot.api.channels);
-
-    return bot.api.channels.repliesAsync({
-      'token': bot.config.incoming_webhook.token,
-      'channel': channel,
-      'thread_ts': threadTs
-    }).then((data) => {
-      if (data && data.hasOwnProperty('messages')) {
-        var fullText = data.messages.filter((message) => {
-          return message.user === user;
-        })
-        .map((message) => {
-          return message.text;
-        })
-        .join('\n');
-        return fullText;
-      }
-      return null;
+    bot.replyWithDialog(message, dialog.asObject(), (err, res) => {
+      console.log(err, res);
```

Comment:
This will print to the console each time the `replyWithDialog` method call resolves. You might want to wrap this in `if (err) { ... }` to ensure you only log on error.